pages/[walletAddress]/fund-wallet.tsx

```diff
@@ -1,13 +1,69 @@
+import { NetworkType } from "@airgap/beacon-sdk";
+import BigNumber from "bignumber.js";
 import { useRouter } from "next/router";
-import { useContext, useEffect } from "react";
+import { useContext, useEffect, useRef, useState } from "react";
+import renderError from "../../components/formUtils";
 import Meta from "../../components/meta";
 import TopUp from "../../components/topUpForm";
+import { TZKT_API_URL, PREFERED_NETWORK } from "../../context/config";
 import { AppDispatchContext, AppStateContext } from "../../context/state";
+import { makeWertWidget } from "../../context/wert";
+import { mutezToTez } from "../../utils/tez";
 
 const TopUpPage = () => {
   const state = useContext(AppStateContext)!;
   const disptach = useContext(AppDispatchContext)!;
   const router = useRouter();
+  const [error, setError] = useState<string | undefined>();
+
+  const onSuccess = async (txId: string) => {
+    if (!state.currentContract) return;
+
+    try {
+      const transaction = await fetch(
+        `${TZKT_API_URL}/v1/operations/${txId}`
+      ).then(res => res.json());
+
+      if (!transaction || transaction.length === 0) {
```

Comment:
need to check the block's level. at least, need to already pass one block/confirmation.